test/controllers/secure_controller_test.js

```diff
@@ -164,21 +157,63 @@ describe('secure controller', function () {
         })
       })
 
-      describe('but the token has been used', function () {
-        it('should display the generic error page', function () {
-          requireSecureController(mockCharge.withSuccess({
+      describe('and the token has been used and the frontend_state cookie is empty', function () {
```

Comment:
This is for checking it's being empty

Reply:
I'll changed this to `and the token has been used and the frontend state cookie is empty`. I didn't how to fit `“cookie” without “frontend_state”` with the preceding `and the token has been used`.

Reply:
Sorry, I should have been clearer. What I meant is that the test is checking what happens when the payment ID isn’t in the cookie. Now, it just so happens that the way it does this is by sending an empty cookie but that’s not the most important detail for this scenario (it could equally send a cookie with other payment IDs in it).

For your other point, “frontend state cookie” is fine (as long as it doesn’t use a specific cookie name like “frontend_state” or “frontend_state_2”).

Reply:
Ah I see. I think this is worth a test scenario.